mozdownload/scraper.py

```diff
@@ -265,6 +311,7 @@ def total_seconds(td):
 
         self.logger.info('Downloading from: %s' % self.url)
         self.logger.info('Saving as: %s' % self.filename)
+        self.logger.info('Directory: %s' % os.path.split(self.url)[0])
```

Comment:
Yes, that's what I meant. Maybe we could also pass the stream to the `download` method, which will use it to store the data inside? This can then be a file output stream, or in case of a checksum an in-memory stream.